lib/bambora/client.rb

```diff
@@ -1,47 +1,107 @@
+    #     c.profiles_api_key = ENV.fetch('BAMBORA_PROFILES_API_KEY')
+    #     c.payments_api_key = ENV.fetch('BAMBORA_PAYMENTS_API_KEY')
+    #     c.batch_api_key = ENV.fetch('BAMBORA_BATCH_API_KEY')
+    #     c.reporting_api_key = ENV.fetch('BAMBORA_REPORTING_API_KEY')
+    #   end
+    #
+    # @param options[base_url] [String] Bambora Base URL
+    # @param options[merchant_id] [String] The Merchant ID for this request.
+    # @param options[sub_merchant_id] [String] The Sub-Merchant ID for this request.
+    # @param options[profiles_api_key] [String] The Profiles API Key for this request.
+    # @param options[payments_api_key] [String] The Payments API Key for this request.
+    # @param options[batch_api_key] [String] The Batch API Key for this request.
+    # @param options[reporting_api_key] [String] The Reporting API Key for this request.
     def initialize(options = {})
-      unless options[:version].nil?
-        raise Bambora::Error, 'Only V1 endpoints are supported at this time.' if options[:version].upcase != 'V1'
+      if !options[:version].nil? && options[:version].upcase != 'V1'
+        raise Bambora::Client::Error, 'Only V1 endpoints are supported at this time.'
       end
 
       options.each do |key, value|
         instance_variable_set("@#{key}", value)
       end
+
       yield(self) if block_given?
     end
 
-    protected
-
-    def get(path:, params:, headers:)
-      connection.get(path, params, headers)
+    # Retrieve a client to make requests to the Profiles endpoints.
+    #
+    # @example
+    #   profiles = client.profiles
+    #   profiles.delete(customer_code: '02355E2e58Bf488EAB4EaFAD7083dB6A')
+    #
+    # @param api_key [String] optional API key for the profiles endpoint. If the Client class was initialized with an
+    #   API key, this parameter is not needed.
+    #
+    # @return [Bambora::V1::ProfileResource]
+    def profiles(api_key: profiles_api_key)
+      @profiles ||= Bambora::V1::ProfileResource.new(client: json_client, api_key: api_key)
     end
 
-    def post(path:, body:, headers:)
-      connection.post(path, body, headers)
+    # Retrieve a client to make requests to the Payments endpoints.
+    #
+    # @example
+    #   payments = client.profiles
+    #   payments.create(
+    #     {
+    #       amount: 50,
+    #       payment_method: 'card',
+    #       card: {
+    #         name: 'Hup Podling',
+    #         number: '4504481742333',
+    #         expiry_month: '12',
+    #         expiry_year: '20',
+    #         cvd: '123',
+    #       },
+    #     },
+    #   )
+    #
+    # @param api_key [String] optional API key for the payments endpoint. If the Client class was initialized with an
+    #   API key, this parameter is not needed.
+    #
+    # @return [Bambora::V1::PaymentResource]
+    def payments(api_key: payments_api_key)
+      @payments ||= Bambora::V1::PaymentResource.new(client: json_client, api_key: api_key)
```

Comment:
Ah, I was confused about this, I thought you namespaced all the files under `/client`. I see now, that this class still lives in `/bambora/v1/...`.